base_layer/core/src/chain.rs

```diff
@@ -119,7 +119,7 @@ impl Chain {
     {
         // The searched hash should always be in the orphan list
         unorphaned_blocks.push(block_hash.clone()); // save all orphans we have used
-        let block = self.orphans.get(block_hash).unwrap().clone(); // todo remove this clone
+        let block = &self.orphans[block_hash].clone(); // todo remove this clone
```

Comment:
This clone might not be necessary after this change

Reply:
Removing the clone results in a borrowing problem.